app/src/main/java/com/github/kulebin/myfoursquareapp/VenueDummyDataSource.java

```diff
@@ -0,0 +1,41 @@
+package com.github.kulebin.myfoursquareapp;
+
+import com.github.kulebin.myfoursquareapp.thread.ITask;
+import com.github.kulebin.myfoursquareapp.thread.IThreadManager;
+import com.github.kulebin.myfoursquareapp.thread.OnResultCallback;
+import com.github.kulebin.myfoursquareapp.thread.ProgressCallback;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class VenueDummyDataSource implements EntityGateway {
+
+    @Override
+    public void fetchVenueList(final OnResultCallback pOnResultCallback) {
+        ((IThreadManager) ContextHolder.get().getSystemService(IThreadManager.APP_SERVICE_KEY)).execute(
```

Comment:
I want not to care where instance of implementation is stored in code.
You can introduce IThreadManager.Impl.get() method for it